src/Host/Client/Impl/BrokerServices/RemoteUriWebService.cs

```diff
@@ -13,13 +13,23 @@
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.WebSockets.Protocol;
 using Microsoft.Common.Core;
+using Microsoft.Common.Core.Logging;
+using Microsoft.Common.Core.Services;
 using Microsoft.R.Host.Protocol;
 using static System.FormattableString;
 
 namespace Microsoft.R.Host.Client.BrokerServices {
     public class RemoteUriWebService : IRemoteUriWebService {
-        public RemoteUriWebService(string baseUri) {
+        private ICoreServices _services;
+        private IConsole _console;
+
+        private readonly IActionLog Log;
+
+        public RemoteUriWebService(string baseUri, ICoreServices services, IConsole console) {
             PostUri = new Uri(new Uri(baseUri), new Uri("/remoteuri", UriKind.Relative));
+            _services = services;
+            _console = console;
+            Log = _services.Log;
```

Comment:
field name should be `_log`, and I think `_services` and `_console` can also be read-only.